Code review

eaarranz opened the pull request.

MarcosIraola committed:
Cambio nombre de frontend en package

MarcosIraola committed:
Subida de pinterest hasta ahora (BE. db, rutas, controllers, clases y mas)(FE. componentes y pages varias)

MarcosIraola committed:
Cargada de perfilUser al hacer click en el boton login sin tener que refrescar. Token guardado en el localstorage, y tambien la carpeta utils con 3 funciones para el token. Agregado el boton y la funcion de logout. Agregado el formulario de registro.

MarcosIraola committed:
importado de react-router-dom, armado de Router en app con switch y sus respectivos links en las paginas.

MarcosIraola committed:
Agregado del file 'routes' para poner la direccion de los links ahi dentro. Actualizado el llamado a los links.

eaarranz commented on backend-api/app/Http/Controllers/SearchController.php:
```diff
@@ -0,0 +1,18 @@
+
+use App\Pin;
+
+class SearchController extends Controller
+{
+    public function searchPinsByNoteAndColor ($query = ''){
+        $pins = Pin::where('color', 'LIKE', '%' . $query . '%')
+            ->orWhere('note', $query)
+            ->get();
+        return response()->json($pins);
+    }
+
+}
```
Soy partidario de que si el search es sobre la entidad Pin, que esté dentro del PinController. Sería añadir un método `search` en el `PinController`

MarcosIraola replied:
Genial. Lo acomodo de esa manera. Lo hice esa clase con Facundo y lo deje asi. En el frontend no lo implemento. Un desafio para un poco mas adelante creo.

eaarranz commented on backend-api/app/Http/Controllers/UserController.php:
```diff
@@ -0,0 +1,80 @@
+        $userValidator = Validator::make($request->all(),[
+            'first_name'=>['required', 'string', 'max:255'],
+            'last_name'=>['required', 'string', 'max:255'],
+            'email'=>['required', 'string', 'max:255'],
```
Puedes usar la validación de email de laravel `email` https://laravel.com/docs/7.x/validation#rule-email

eaarranz commented on backend-api/config/mail.php:
```diff
@@ -107,7 +107,7 @@
     |
     | If you are using Markdown based email rendering, you may configure your
-    | theme and component paths here, allowing you to customize the design
+    | theme and Calculadora paths here, allowing you to customize the design
```
ups!

MarcosIraola replied:
Esto no se que es! En el proyecto deje guardados en otro file otros ejercicios , como el de la calculadora y la primera prueba de un pinterest.

eaarranz commented on backend-api/routes/api.php:
```diff
@@ -13,6 +14,66 @@
+Route::get('test', function (){
+   Log::info( 'TEST recived');
+   return'Hello World';
 });
```
Ups!

eaarranz commented on backend-api/routes/api.php:
```diff
@@ -13,6 +14,66 @@
+|--------------------------------------------------------------------------
+*/
+
+Route::get('users', 'UserController@all');
+Route::get('users/{id}', 'UserController@getById');
+Route::get('users/email/{email}', 'UserController@getByEmail');
+Route::get('users/username/{username}', 'UserController@getByUsername');
+Route::post('users', 'UserController@create');
+Route::put('users/{id}', 'UserController@updateById');
+Route::delete('users/{id}', 'UserController@delete');
```
Puedes usar los grupos aquí. Todos empiezan por `users`.
`Route::group([

    'middleware' => 'users',
    'prefix' => 'auth'

], function ($router) {`

MarcosIraola replied:
Si! Ahora lo acomodo.

eaarranz commented on frontend-app/src/OtrosEjercicios/Calculadora/Button/Button.js:
```diff
@@ -0,0 +1,18 @@
+
+    const styleColor = {
+        backgroundColor: color,
+    }
```
You can use css modules here. It's an advanced use case but you can use: classNames to dynamically set a class here. https://github.com/JedWatson/classnames

eaarranz commented on frontend-app/src/OtrosEjercicios/Pinterest/Pinterest.js:
```diff
@@ -0,0 +1,45 @@
+        <div className={'contenedor__pinterest'}>
+
+            <div className={'galeria__home'}>
+                <img src={'https://images.unsplash.com/photo-1592819005124-1401b90865dd?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=750&q=80'} className={'img__galeria'}/>
+                <img src={'https://images.unsplash.com/photo-1592838981793-f3ebf267056d?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=750&q=80'} className={'img__galeria'}/>
+                <img src={'https://images.unsplash.com/photo-1592840331052-16e15c2c6f95?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=334&q=80'} className={'img__galeria'}/>
+                <img src={'https://images.unsplash.com/photo-1592771404380-467f535c7c4f?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=334&q=80'} className={'img__galeria'}/>
+                <img src={'https://images.unsplash.com/photo-1592842825197-89dd8efba31f?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=334&q=80'} className={'img__galeria'}/>
+                <img src={'https://images.unsplash.com/photo-1592819649132-7838ae2b130c?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=750&q=80'} className={'img__galeria'}/>
+                <img src={'https://images.unsplash.com/photo-1592838464221-a7268248a9ec?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=401&q=80'} className={'img__galeria'}/>
+                <img src={'https://images.unsplash.com/photo-1592838919349-583a8fa23659?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=750&q=80'} className={'img__galeria'}/>
+                <img src={'https://images.unsplash.com/photo-1592763945926-1c89c58fa90f?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=334&q=80'} className={'img__galeria'}/>
+                <img src={'https://images.unsplash.com/photo-1592819005124-1401b90865dd?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=750&q=80'} className={'img__galeria'}/>
+                <img src={'https://images.unsplash.com/photo-1592822347747-2e548b2aca50?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=334&q=80'} className={'img__galeria'}/>
+                <img src={'https://images.unsplash.com/photo-1592840331052-16e15c2c6f95?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=334&q=80'} className={'img__galeria'}/>
+                <img src={'https://images.unsplash.com/photo-1592819005124-1401b90865dd?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=750&q=80'} className={'img__galeria'}/>
+                <img src={'https://images.unsplash.com/photo-1592838981793-f3ebf267056d?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=750&q=80'} className={'img__galeria'}/>
+                <img src={'https://images.unsplash.com/photo-1592771404380-467f535c7c4f?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=334&q=80'} className={'img__galeria'}/>
+                <img src={'https://images.unsplash.com/photo-1592842825197-89dd8efba31f?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=334&q=80'} className={'img__galeria'}/>
+                <img src={'https://images.unsplash.com/photo-1592819649132-7838ae2b130c?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=750&q=80'} className={'img__galeria'}/>
+                <img src={'https://images.unsplash.com/photo-1592838464221-a7268248a9ec?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=401&q=80'} className={'img__galeria'}/>
+                <img src={'https://images.unsplash.com/photo-1592838919349-583a8fa23659?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=750&q=80'} className={'img__galeria'}/>
+                <img src={'https://images.unsplash.com/photo-1592763945926-1c89c58fa90f?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=334&q=80'} className={'img__galeria'}/>
+                <img src={'https://images.unsplash.com/photo-1592822347747-2e548b2aca50?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=334&q=80'} className={'img__galeria'}/>
+                <img src={'https://images.unsplash.com/photo-1592842825197-89dd8efba31f?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=334&q=80'} className={'img__galeria'}/>
+                <img src={'https://images.unsplash.com/photo-1592819649132-7838ae2b130c?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=750&q=80'} className={'img__galeria'}/>
```
Use a pins.map(pin => <img ... />) here. You can use an array with the images and it will look a loooot cleaner.

MarcosIraola replied:
Esto lo mismo que respondi en otro comentario. Es la primera prueba del pinterest, que escribi varias imagenes de esta manera para poder acomodarlas en la home, antes de que traigamos las imagenes de nuestra API.

eaarranz reviewed (commented):
Primera parte

MarcosIraola committed:
Agregado un redirect a PERFILUSER cuando se hace click en el boton de login. Tmb un redirect a LOGIN cuando se hace click en REGISTER (un nuevo usuario)

MarcosIraola committed:
Agrupado de rutas para la API, con route::group s. Eliminado de proyectos viejos como 'CALCULADORA' y 'PINTEREST'. Modificacion de lugar de SEARCH by QUERY de pins, y puesto en PinController. Eliminacion de SEARCHCONTROLLER

MarcosIraola committed:
-NavBar diferente si hay authentication o no.
-BoardForm no necesita as user id -> lo trae desde auth()->me()->id (modificado en backend)
-Landing Page creada

MarcosIraola committed:
Landing page con botones de login y register / no acceso al resto de la web

MarcosIraola committed:
Boton de Cerrar formLogin y formRegister

MarcosIraola committed:
Creado BoardCard y ListBoard. Implementado aunque board-id esta hardcodeado en sus fetchs.

MarcosIraola committed:
Authentication con context agregado. Private route agregado a PINBOARDFORM y funciona.

MarcosIraola committed:
Agregado de PrivateRoutes y cambiado por las Routes donde corresponde. Eliminacion de funcion isAuthenticated vieja de localStorage.utils

MarcosIraola committed:
Eliminacion del fetch en perfiluser para traer los datos del usuario logeado. Ahora se hace desde el state.user.{} por context. Modificacion de la url hardcodeada para traer los listboars, ahora llama a los listboards del usuario logead.

MarcosIraola committed:
alt={'Error'} agregado en todas las etiquetas <img> salvo en boardCard.

MarcosIraola committed:
agregado un par mas de Alts a algunas imagenes. Eliminacion de codigo que se importaba y no se usaba

MarcosIraola committed:
Agregado de columna AVATAR en tabla users. Modificacion en perfiluser para rellenar la img con la del user logeado

MarcosIraola committed:
Eliminacion de las tablas extras de user (las llamadas avatar_url y avatarl_url2). Eliminacion de varios errores que marcaba el lint {}

MarcosIraola committed:
Create .travis.yml

MarcosIraola committed:
Merge remote-tracking branch 'origin/master'

MarcosIraola committed:
Eliminacion de las tablas extras de user (las llamadas avatar_url y avatarl_url2). Eliminacion de varios errores que marcaba el lint {}

MarcosIraola committed:
travis

MarcosIraola committed:
travis

MarcosIraola committed:
travis
